src/passport-saml/saml.ts

```diff
@@ -628,24 +654,24 @@ class SAML {
 
     const signature = signatures[0];
     return certs.some(certToCheck => {
-      return this.validateSignatureForCert(signature, certToCheck, fullXml, currentNode);
+      return this.validateSignatureForCert(signature as string, certToCheck, fullXml, currentNode);
     });
   }
 
 // This function checks that the |signature| is signed with a given |cert|.
-  validateSignatureForCert(signature, cert, fullXml, currentNode) {
+  validateSignatureForCert(signature: string | Node, cert: string, fullXml: string, currentNode: HTMLElement) {
     const sig = new xmlCrypto.SignedXml();
     sig.keyInfoProvider = {
-      file: undefined,
+      file: '',
```

Comment:
Is this change really correct? Shouldn't it be `null` or simply not defined?

Reply:
keyInfoProvider requires a string here as input (but does not use it)